master/buildbot/util/git.py

```diff
@@ -270,53 +345,344 @@ def checkFeatureSupport(self):
+
+    def _adjust_command_params_for_ssh_private_key(
+        self,
+        full_command: list[str],
+        full_env: dict[str, str],
+        workdir: str,
+        git_mixin: GitMixin,
+    ) -> None:
+        key_path = self._get_ssh_private_key_path(workdir)
+        host_key_path = None
+        if self.ssh_host_key is not None or self.ssh_known_hosts is not None:
+            host_key_path = self._get_ssh_host_key_path(workdir)
+
+        ssh_wrapper_path = self._get_ssh_wrapper_script_path(workdir)
+
+        git_mixin.adjustCommandParamsForSshPrivateKey(
+            full_command, full_env, key_path, ssh_wrapper_path, host_key_path,
+        )
+
+    def adjust_git_command_params_for_auth(
+        self,
+        full_command: list[str],
+        full_env: dict[str, str],
+        workdir: str,
+        git_mixin: GitMixin,
+    ) -> None:
+        self._adjust_command_params_for_ssh_private_key(
+            full_command, full_env,
+            workdir=workdir, git_mixin=git_mixin,
+        )
 
-        private_key_path = self._getSshPrivateKeyPath(ssh_data_path)
+
+    @abc.abstractmethod
+    @defer.inlineCallbacks
+    def _download_file(
+        self, path: str, content: str, mode: int, workdir: str | None = None,
+    ) -> Generator[defer.Deferred[Any], Any, None]:
+        raise NotImplementedError()
+
+    @defer.inlineCallbacks
+    def download_auth_files_if_needed(
+        self,
+        workdir: str,
+        download_wrapper_script: bool = False,
+    ) -> Generator[defer.Deferred[Any], Any, int]:
+        if self.ssh_private_key is None:
+            return RC_SUCCESS
+
+        p = Properties()
+        p.master = self._master
+        private_key = yield p.render(self.ssh_private_key)
+        host_key = yield p.render(self.ssh_host_key)
+        known_hosts = yield p.render(self.ssh_known_hosts)
+
+        private_key_path = self._get_ssh_private_key_path(workdir)
         private_key = ensureSshKeyNewline(private_key)
-        yield self.downloadFileContentToWorker(
-            private_key_path, private_key, workdir=workdir, mode=0o400
+        yield self._download_file(
+            private_key_path, private_key, mode=stat.S_IRUSR, workdir=workdir,
         )
 
         known_hosts_path = None
-        if self.sshHostKey is not None or self.sshKnownHosts is not None:
-            known_hosts_path = self._getSshHostKeyPath(ssh_data_path)
+        if self.ssh_host_key is not None or self.ssh_known_hosts is not None:
+            known_hosts_path = self._get_ssh_host_key_path(workdir)
 
-            if self.sshHostKey is not None:
+            if known_hosts is not None:
+                known_hosts_contents = known_hosts
+            else:
                 known_hosts_contents = getSshKnownHostsContents(host_key)
-            yield self.downloadFileContentToWorker(
-                known_hosts_path, known_hosts_contents, workdir=workdir, mode=0o400
+
+            yield self._download_file(
```

Comment:
Maybe keep name _download_file_content_to_worker?

Reply:
Wouldn't that be confusing in the context of `GitServiceAuth` which has no worker?

Reply:
Indeed, I'm wrong.